### DISCUSSION TITLE
Problems with starting the project

### DISCUSSION THREAD

**Discussion by nahary** [openkfw/TruBudget#1309] [General]
Hello guys,

We used docker for pulling your repo and we run it (**docker pull trubudget/blockchain**)
Then we started containers, but localhost:3000 didn't work.
We tried to start it in different ports but it didn't help.
Please give us advice on how can we solve this problem.

**Comment by georgimld** [comment]
Hi,

in order for TruBudget to start and the UI to be available you need to start (at least) 3 containers: **blockchain**, **api** and **frontend**. They also need certain environement variables to be set in order to be able to connect to each other.

There are a few ways to achieve this:

- as mentioned here https://github.com/openkfw/TruBudget/tree/main/docker-compose you can use the docker-compose scripts in the docker-compose folder (e.g. the command `docker-compose --project-directory . -f blockchain/docker-compose.alphanode.yml -f api/docker-compose.yml -f frontend/docker-compose.yml up` - personally i prefer this option
- another option is with the operation script as mentioned here https://github.com/openkfw/TruBudget/tree/main/scripts/operation

If you still have errors there are a few things you can check:

- logs of the containers (no error logs or logs that the connection is successful)
- check the network tab in the UI for the reqeusts being sent to the api - the result usually tells you if it's an internal problem with the api/credentials or if it's a setup problem

Hope this helps, let me know if you have any more questions 😊

**Reply by nahary** [reply]
Thank you for your reply, we restarted all containers via script and now we have **user.authenticate** error, we didn't change .env,  so we tried root/root-secret but it didn't work.
![image](https://user-images.githubusercontent.com/30175768/219553829-4267d843-89ac-42d1-93cf-52626aa42c7b.png)

**Reply by georgimld** [reply]
Sorry for the late reply, please check the api and blockchain logs, it seems that you are getting a 500 error (internal server error), so there's probably still a problem with the setup. 

you could check the following things:
- is the chain running? (you should see a "Node running" log in the blockchain container)
- can the api connect to the blockchain? (the api would display error logs if it can't reach the blockchain)
